internal/flux_controller.go

```diff
@@ -14,9 +15,12 @@ type FluxControllerMock struct {
 func NewFluxControllerMock() *FluxControllerMock {
 	return &FluxControllerMock{
 		QueryFn: func(ctx context.Context, compiler flux.Compiler) (query flux.Query, e error) {
-			// See https://github.com/influxdata/influxdb/issues/14161
-			//return NewFluxQueryMock(), nil
-			return nil, nil
+			p, err := compiler.Compile(ctx)
+			if err != nil {
+				return nil, err
+			}
+			alloc := &memory.Allocator{}
+			return p.Start(ctx, alloc)
```

Comment:
Can't you adapt `FluxQueryMock` to implement `flux.Query` and return that instead?

Reply:
I thought it was easier to just have the default use the compiler. The purpose of this is to make it possible to override the query function for the controller I believe.